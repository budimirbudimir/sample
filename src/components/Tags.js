// @flow

import React from 'react'

import type { Tag } from '../models'

type TagsTypes = {
	// tags is array of objects containing certain tag data
	tags: Tag[],
}

const Tags = ({ tags }: TagsTypes) => (
	<p>
		Tags:
		{tags.map((tag: Tag, index) => (
			<a key={index} className="App-similar_act" href={tag.url}>
				{tag.name}
			</a>
		))}
	</p>
)

export default Tags
