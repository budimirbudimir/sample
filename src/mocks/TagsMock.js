export default [
	{ name: 'rock', url: 'https://www.last.fm/tag/rock' },
	{ name: 'pop', url: 'https://www.last.fm/tag/pop' },
	{ name: 'metal', url: 'https://www.last.fm/tag/metal' },
	{ name: 'disco', url: 'https://www.last.fm/tag/disco' },
	{ name: 'reggae', url: 'https://www.last.fm/tag/reggae' },
]
