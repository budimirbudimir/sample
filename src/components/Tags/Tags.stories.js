import React from 'react'

import { storiesOf } from '@storybook/react'

import Tags from '.'
import '../styles/App.css'

const tagList = [
  { name: 'rock', url: 'https://www.last.fm/tag/rock' },
  { name: 'pop', url: 'https://www.last.fm/tag/pop' },
  { name: 'metal', url: 'https://www.last.fm/tag/metal' },
  { name: 'disco', url: 'https://www.last.fm/tag/disco' },
  { name: 'reggae', url: 'https://www.last.fm/tag/reggae' }
]

storiesOf('Tags', module).add('Default state', () => <Tags tags={tagList} />)
