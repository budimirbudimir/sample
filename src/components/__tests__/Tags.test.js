import React from 'react'
import Tags from '../Tags'
import { shallow } from 'enzyme'

import TagsMock from '../../mocks/TagsMock'

describe('Tags', () => {
	it('matches snapshot', () => {
		const wrapper = shallow(<Tags tags={TagsMock} />)
		expect(wrapper).toMatchSnapshot()
	})
})
