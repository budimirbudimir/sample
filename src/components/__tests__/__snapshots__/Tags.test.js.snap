// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    tags={
      Array [
        Object {
          "name": "rock",
          "url": "https://www.last.fm/tag/rock",
        },
        Object {
          "name": "pop",
          "url": "https://www.last.fm/tag/pop",
        },
        Object {
          "name": "metal",
          "url": "https://www.last.fm/tag/metal",
        },
        Object {
          "name": "disco",
          "url": "https://www.last.fm/tag/disco",
        },
        Object {
          "name": "reggae",
          "url": "https://www.last.fm/tag/reggae",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="Tags-title"
        >
          Tags:
        </span>,
        Array [
          <a
            className="App-similar_act"
            href="https://www.last.fm/tag/rock"
          >
            rock
          </a>,
          <a
            className="App-similar_act"
            href="https://www.last.fm/tag/pop"
          >
            pop
          </a>,
          <a
            className="App-similar_act"
            href="https://www.last.fm/tag/metal"
          >
            metal
          </a>,
          <a
            className="App-similar_act"
            href="https://www.last.fm/tag/disco"
          >
            disco
          </a>,
          <a
            className="App-similar_act"
            href="https://www.last.fm/tag/reggae"
          >
            reggae
          </a>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tags:",
          "className": "Tags-title",
        },
        "ref": null,
        "rendered": "Tags:",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "rock",
          "className": "App-similar_act",
          "href": "https://www.last.fm/tag/rock",
        },
        "ref": null,
        "rendered": "rock",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "pop",
          "className": "App-similar_act",
          "href": "https://www.last.fm/tag/pop",
        },
        "ref": null,
        "rendered": "pop",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "metal",
          "className": "App-similar_act",
          "href": "https://www.last.fm/tag/metal",
        },
        "ref": null,
        "rendered": "metal",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "disco",
          "className": "App-similar_act",
          "href": "https://www.last.fm/tag/disco",
        },
        "ref": null,
        "rendered": "disco",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": "reggae",
          "className": "App-similar_act",
          "href": "https://www.last.fm/tag/reggae",
        },
        "ref": null,
        "rendered": "reggae",
        "type": "a",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="Tags-title"
          >
            Tags:
          </span>,
          Array [
            <a
              className="App-similar_act"
              href="https://www.last.fm/tag/rock"
            >
              rock
            </a>,
            <a
              className="App-similar_act"
              href="https://www.last.fm/tag/pop"
            >
              pop
            </a>,
            <a
              className="App-similar_act"
              href="https://www.last.fm/tag/metal"
            >
              metal
            </a>,
            <a
              className="App-similar_act"
              href="https://www.last.fm/tag/disco"
            >
              disco
            </a>,
            <a
              className="App-similar_act"
              href="https://www.last.fm/tag/reggae"
            >
              reggae
            </a>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tags:",
            "className": "Tags-title",
          },
          "ref": null,
          "rendered": "Tags:",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "rock",
            "className": "App-similar_act",
            "href": "https://www.last.fm/tag/rock",
          },
          "ref": null,
          "rendered": "rock",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "pop",
            "className": "App-similar_act",
            "href": "https://www.last.fm/tag/pop",
          },
          "ref": null,
          "rendered": "pop",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "metal",
            "className": "App-similar_act",
            "href": "https://www.last.fm/tag/metal",
          },
          "ref": null,
          "rendered": "metal",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "disco",
            "className": "App-similar_act",
            "href": "https://www.last.fm/tag/disco",
          },
          "ref": null,
          "rendered": "disco",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": "reggae",
            "className": "App-similar_act",
            "href": "https://www.last.fm/tag/reggae",
          },
          "ref": null,
          "rendered": "reggae",
          "type": "a",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
